src/subcommand/init.rs

```diff
@@ -244,6 +275,26 @@ function _zoxide_hook --on-variable PWD
 end
 "#;
 
+// FIXME(cole-h): This is hacky, but the only way I was able to get it to work.
+const POSIX_HOOK_PWD: &str = r#"
+_zoxide_hook() {
+    touch /tmp/.zo_pwd
```

Comment:
A filename like this has problems:

- Multiple users will overwrite each others tmp files
- Multiple shell instances will overwrite each others tmp files
- `/tmp` is not always the preferred tmp directory

`mktemp` will create a tmp file for us in a compliant way, that won't clash with any other tmp file. We can this to generate a new file for each shell instance. However, `mktemp` is not part of the POSIX standard. While it is present on most modern operating systems, I still feel that using it defeats the purpose of POSIX compliance.

We could write our own logic for generating a tmp filename. However, we will end up cluttering the users tmp directory in no time, since every new shell will create a new tmp file - so we will also need to clean up upon shell exit by trapping `EXIT`, `INT`, `HUP`, and `TERM`.

It seems we really have opened a can of worms here - it may make sense to leave out the PWD hook as something we won't support in POSIX shells.

Reply:
(Purposefully not marking this as "resolved" so anybody who comes by here looking for answers as to why we didn't add a PWD hook can see the reasoning without needing to expand all the previously-resolved reviews.)

Reply:
I've added a PWD hook for POSIX shells in https://github.com/ajeetdsouza/zoxide/commit/7532fe98d88532022a95d387c4d21928b7026336, so I'm marking this as resolved.